crates/datafusion_ext/src/vars/inner.rs

```diff
@@ -0,0 +1,370 @@
+use datafusion::arrow::array::StringArray;
+use datafusion::arrow::datatypes::{DataType, Field, Schema};
+use datafusion::arrow::record_batch::RecordBatch;
+use datafusion::config::ConfigEntry;
+use datafusion::error::Result;
+use datafusion::variable::VarType;
+use std::borrow::Borrow;
+
+use super::constants::*;
+use super::error::VarError;
+use super::value::Value;
+use std::sync::Arc;
+use tracing::error;
+use uuid::Uuid;
+
+/// Variables for a session.
+#[derive(Debug)]
+pub struct SessionVarsInner {
```

Comment:
this is mostly unchanged from the original `SessionVars`.

the new `SessionVars` is a wrapper around this that provides some read/write utilities around this inner struct.